### PR TITLE
Add recipe for CARNAC-LR

### PR DESCRIPTION
**Pull request by npavlovikj**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by npavlovikj** [comment]
@bioconda/osx , can someone please help me with the OSX build? I am getting:
```
12:47:19 BIOCONDA INFO (ERR) Undefined symbols for architecture x86_64:
12:47:19 BIOCONDA INFO (ERR)   "___kmpc_barrier", referenced from:
12:47:19 BIOCONDA INFO (ERR)       _.omp_outlined. in clustering_cliqueness.o
12:47:19 BIOCONDA INFO (ERR)       _.omp_outlined..20 in clustering_cliqueness.o
...
```
errors.

The `CARNAC-LR` authors say that the OSX version doesn't work with `-fopenmp` and multiple threads. I tried with and without `llvm-openmp`, and modifying `LDFLAGS`, `CFLAGS` and `DYLD_LIBRARY_PATH` according similar recipes locaclly, but the OSX errors persist.

I would appreciate any help and suggestions!

**Comment by acaprez** [comment]
If you look at the build output from both linux and OS X, it's not actually honoring the `CFLAGS` variable from the environment.  It's just that the linux build works even with the one from the makefile.  You need to pass `CFLAGS` on the `make` line to have it use the value from the environment, i.e. do `make CC=$CC CXX=$CXX LDFLAGS="$LDFLAGS" CFLAGS="$CFLAGS"`

Note the anaconda compilers set `-std=c++14` in `CFLAGS`, which broke the compile, at least on OS X.  To get it to build, I had to do `export CFLAGS="$CFLAGS -std=c++11"` in the Darwin if block in the build script.  By putting `-std=c++11` _after_ all the other flags, it will use that instead of c++14.  I suspect you'll have to do the same thing on linux, but I didn't check that.

**Comment by npavlovikj** [comment]
@acaprez , thank you so much - your suggestions fixed the issue!
Both builds passed now, so if you don't mind, can you please review the recipe at your convenience as well?

File: recipes/carnac-lr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  run:
+    - python >=3
+
+test:
+  files:
```
Comment (bgruening) [inline recipes/carnac-lr/meta.yaml]: is this file now needed?

Comment (npavlovikj) [reply recipes/carnac-lr/meta.yaml]: The test file or the makefile patch?

Comment (bgruening) [reply recipes/carnac-lr/meta.yaml]: the test file...

Comment (npavlovikj) [reply recipes/carnac-lr/meta.yaml]: The test file is needed for the test in run_test.sh ("CARNAC-LR -f test.input -o output.txt
"). Should I not include test at all? The file is 3.2KB, so I thought having the test is a good practice?

**Comment by npavlovikj** [comment]
@bgruening , I am sorry for pinging you here, but can you please let me know if you have any additional comments I need to address before this PR is approved for merging?

**Comment by npavlovikj** [comment]
Thank you so much for the merge @dpryan79  - I really appreciate it!

**Pull request merged**